### PR TITLE
(#273) Fix footer headers acting as links

### LINKED ISSUE #270: Footer headers incorrectly act as links
### Issue description
This may be a footer component issue, or just an issue with the example. The NCI Big, and the normal "big", example on the Footer Component page, https://designsystem-dev.cancer.gov/develop/components/footer, does not display correctly.

Issues:
1. Hovering over the heading shows the finger pointer.
2. Clicking on the heading causes it to be highlighted and take focus.
    See image
2. Clicking on the heading cause the links underneath, and elsewhere in the footer, to expand and collapse (it is too subtle to appear in the screen shot. 
     * Expanded
        image
     * Collapsed
        image
3. Clicking on the heading causes all other links to flash into a focus state.

### PR DESCRIPTION
**Pull request by olitharp-nci**
Closes #270

### CODE REVIEW

**Comment by olitharp-nci** [comment]
https://designsystem-dev.cancer.gov/pr-272/components/footer

### Notes

Adding `onClick` to a header:
<ol>
  <li>ESLint: Non-interactive elements should not be assigned mouse or keyboard event listeners.(jsx-a11y/no-noninteractive-element-interactions) (aka, headers should not be interactive, warning is removed with role=presentation but role should be button on small screens)</li>
  <li> ESLint: Visible, non-interactive elements with click handlers must have at least one keyboard listener.(jsx-a11y/click-events-have-key-events) (aka: duplicate onClick scripts with onKeyDown )</li>
</ol> 

Adding `role=button` to a header: 

<ol start="3">
  <li>ESLint: Non-interactive elements should not be assigned interactive roles.(jsx-a11y/no-noninteractive-element-to-interactive-role) (See note 1)</li>
</ol> 

Using anything other than a button with `role=button`:

<ol start="4">
  <li>ESLint: Elements with the 'button' interactive role must be tabbable.(jsx-a11y/interactive-supports-focus) (aka: adding tabIndex)</li>
  <li>ESLint: Visible, non-interactive elements with click handlers must have at least one keyboard listener.(jsx-a11y/click-events-have-key-events) (See note 2)</li>
</ol> 

`<button><header/></button>` vs `<header><button/></header>`:

<ol start="6">
  <li> <a href="https://designsystem.digital.gov/components/accordion/#usability-guidance-accordion">Accordion Usability Guidance</a>: Make the entire header selectable. Allow users to click anywhere in the header area to expand or collapse the content; a larger target is easier to manipulate.</li>
</ol>

**Review by karlikpj — APPROVED** [review]

File: docs/src/components/footer/nci-big.jsx [hunk 1]
```diff
@@ -24,14 +24,17 @@ export const NCIBigFooter = () => {
 											} `}>
 											<button
 												onClick={() => setActiveItem(activeItem === 1 ? 0 : 1)}
-												className="usa-button usa-button--unstyled">
-												<h4 className="usa-footer__primary-link">
-													Primary link 1
+												className="usa-button usa-button--unstyled usa-footer__nci-accordion-button">
+												<h4 className="usa-footer__primary-link usa-footer__nci-primary-link--accordion-header">
```
Comment (arcepaul) [inline docs/src/components/footer/nci-big.jsx:28]: I don't think headings are allowed inside of buttons.  I'm surprised linting didn't complain.

Comment (olitharp-nci) [reply docs/src/components/footer/nci-big.jsx:28]: do we even need a heading here? styles are all in a class, not attached to any `h*`
could use a span or div instead

Comment (olitharp-nci) [reply docs/src/components/footer/nci-big.jsx]: This has been fixed

File: docs/src/components/__tests__/Footer.test.js [hunk 2]
```diff
@@ -109,48 +109,48 @@ describe('Footer', () => {
...
-		fireEvent.click(screen.getByText('Primary link 1'));
-		await waitFor(() => screen.getByLabelText('Primary link 1'));
-		expect(screen.getByLabelText('Primary link 1')).toHaveTextContent(
+		fireEvent.click(screen.getByRole('button', { name: 'Header 1' }));
+		await waitFor(() => screen.getByLabelText('Header 1'));
+		expect(screen.getByLabelText('Header 1')).toHaveTextContent(
 			'Secondary link 1'
 		);
 
 		// Second accordion activates
```
Comment (arcepaul) [inline docs/src/components/__tests__/Footer.test.js:129]: These following tests are probably overly redundant

Comment (olitharp-nci) [reply docs/src/components/__tests__/Footer.test.js]: :+1: cleaned up tests for `nci-big` and removed script for `big `

**Review by arcepaul — APPROVED** [review]

**Pull request merged**